### PR TITLE
On HiDPI screens with low resolution, allow widgetry applications to

### PR DESCRIPTION
**Pull request by dabreegster**
override the default scale factor in order to achieve a minimum width.

Problem: I keep seeing screenshots from people that look like this:
Screenshot from 2021-02-22 15-40-14
When they force `--scale_factor=1`, things look normal.

Ideal solution: Come up with a proper plan for supporting devices with low width. This is hard, because it requires both redesigning many parts of the UI (sandbox mode panels, dashboard pages, etc) and maintaining multiple layouts in the code.

Workaround: Detect this problem and force scale factor to 1. Retain the ability for people to use the `--scale_factor` flag, if things wind up too small on their screen.

I tested on my device (which can do a factor of 1 or 2). @Robinlovelace has an old Windows laptop that exhibits this problem; I'm guessing compiling this branch there is annoying. If I send you a Windows binary, could you please test it out?
Screenshot from 2021-02-22 15-40-35
Here's the fix on mine -- big cursor and taskbar, "normal" abst.

### CODE REVIEW

File: game/src/lib.rs [hunk 1]
```diff
@@ -44,7 +44,9 @@ pub fn main() {
     let mut settings = widgetry::Settings::new("A/B Street")
         .read_svg(Box::new(abstio::slurp_bytes))
         .window_icon(abstio::path("system/assets/pregame/icon.png"))
-        .loading_tips(map_gui::tools::loading_tips());
+        .loading_tips(map_gui::tools::loading_tips())
+        // This is approximately how much the 3 top panels in sandbox mode require.
```
Comment (dabreegster) [inline game/src/lib.rs:48]: I measured this by printing `panel.contents_dims`. In sandbox mode, I get 332 + 707 + 416 = 1455 for the top (left, center, right) panels in sandbox mode on an empty montlake map.

File: widgetry/src/drawing.rs [hunk 2]
```diff
@@ -246,7 +246,7 @@ pub struct Prerender {
     pub(crate) inner: PrerenderInnards,
     pub(crate) assets: Assets,
     pub(crate) num_uploads: Cell<usize>,
-    pub(crate) scale_factor: RefCell<f64>,
```
Comment (dabreegster) [inline widgetry/src/drawing.rs:249]: Unrelated cleanup while I was squinting carefully at this code again. We don't listen to scale factor changed events from winit (which might happen if you drag a window between monitors), so this value doesn't ever change after init.

File: widgetry/src/screen_geom.rs [hunk 3]
```diff
@@ -139,10 +139,6 @@ impl ScreenDims {
             }
         }
     }
-
-    pub fn scaled(&self, factor: f64) -> ScreenDims {
```
Comment (dabreegster) [inline widgetry/src/screen_geom.rs:143]: Unused

File: game/src/lib.rs [hunk 4]
```diff
@@ -44,7 +44,9 @@ pub fn main() {
...
+        .require_minimum_width(1500.0);
```
Comment (michaelkirk) [inline game/src/lib.rs:49]: I guess this is in units of logical pixels?

**Review by michaelkirk — APPROVED** [review]
LGTM - seems like an appropriately expedient fix.

File: game/src/lib.rs (hunk 4 above)
Comment (dabreegster) [reply game/src/lib.rs:49]: Yes -- good catch! Updated the docs.

**Comment by dabreegster** [comment]
Thanks for the quick review. To double-confirm the fix, I'll ask Robin to compare two web deployments.

**Comment by Robinlovelace** [comment]
> I tested on my device (which can do a factor of 1 or 2). @Robinlovelace has an old Windows laptop that exhibits this problem; I'm guessing compiling this branch there is annoying. If I send you a Windows binary, could you please test it out?

Happy to test it out 👍
